checks/collector.py

```diff
@@ -751,7 +751,7 @@ def _get_hostname_metadata(self):
         except Exception:
             pass
 
-        metadata["hostname"] = get_hostname()
+        metadata["hostname"] = get_hostname(config=self.agentConfig)
```

Comment:
Wow, nice catch, `self.hostname` seems to come from this [commit](https://github.com/DataDog/dd-agent/commit/766ee8f8b495d6d3f61da1fcc5123ad2c413de39), it's probably just an omission. Fixing it.